### PR TITLE
 Modified the interface after the selection group change

### LINKED ISSUE #9108: Add "Home"-Button to entry-editor
When an entry is picked from the table of entries the entry-editor is positioned to this entry.
When a group is selected, NOT containing this entry, the entry-editor keeps his position/content (what else...),
there is NO highlighting in the table of entries any more (what else...).

When a (super-)group is highlighted, again containing the entry of the entry-editor,
there is still NO highlighting in the table of entries. This should be improved, it is contra-intuitive
The corresponding entry in the table of entries should be highlighted again.

A new button "Home" should be added to the in the entry-editor.
It should force the selection of groups to walk the tree backwards until the 
group is containing this entry and the entry should be highlighted.
Alternatively, the group should be switched to "All entries",
in case walking the tree backwards is to costly.

This button allows for jumping back from a "search" in the groups, i.e a tentative
switching between groups.
 
Best, HSvia

### PR DESCRIPTION
**Pull request by zeyu-gong**
Fixes #9108 

At this point, having the program memory entries in the entry editor that are open may involve a lot of refactoring. I think it makes more sense to close the entry editor after you select another group.

Made some changes to try to close the entry editor when reselecting the group， Because I think this seems like a logical solution to issue 9108 , but the code doesn't seem to close the editor.

Hope to get some guidance

- [X] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [X] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [X] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [X] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by ThiloteE** [comment]
To ease organizational workflows I have linked the pull-request to the issue with syntax as described in https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

> <h2 id="linking-a-pull-request-to-an-issue-using-a-keyword">Linking a pull request to an issue using a keyword</h2>
> <p>You can link a pull request to an issue by using a supported keyword 
> in the pull request's description or in a commit message. The pull 
> request <strong>must be</strong> on the default branch.</p>
> <ul>
> <li>close</li>
> <li>closes</li>
> <li>closed</li>
> <li>fix</li>
> <li>fixes</li>
> <li>fixed</li>
> <li>resolve</li>
> <li>resolves</li>
> <li>resolved</li>
> </ul>
> <p>If you use a keyword to reference a pull request comment in another 
> pull request, the pull requests will be linked. Merging the referencing 
> pull request also closes the referenced pull request.</p>
> <p>The syntax for closing keywords depends on whether the issue is in the same repository as the pull request.</p>

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 1]
```diff
@@ -510,13 +503,19 @@ private void createMainTable() {
                 Globals.IMPORT_FORMAT_READER);
 
         // Add the listener that binds selection to state manager (TODO: should be replaced by proper JavaFX binding as soon as table is implemented in JavaFX)
+
         mainTable.addSelectionListener(listEvent -> stateManager.setSelectedEntries(mainTable.getSelectedEntries()));
 
+        // add listener that when main table change close the entry editor
+        mainTable.addSelectionListener(listEvent -> entryEditorClosing());
+
+
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/LibraryTab.java:512]: Why is our reviewdog not complaining about the second emtpy line? @Siedlerchr

**Comment by calixtus** [comment]
How about `EntryEditor::close`?

**Comment by calixtus** [comment]
What's the status here?

**Comment by calixtus** [comment]
Since there has been sadly no reaction for more than 20 days i'm closing this PR, as this in no mergeable state and the chosen solution seems require some refinement. Please feel free to object and create a new PR if you want to continue on this issue.

**Pull request closed without merging**